.github/scripts/delete-old-cloudsmith-artifacts.sh

```diff
@@ -1,42 +1,67 @@
 #!/bin/bash
 # Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
-dnf install -y 'dnf-command(config-manager)' jq
 
-set -euo pipefail
+if [ "${RUNNER_DEBUG:-}" == "1" ]; then
+    set -o xtrace
+fi
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+dnf install -yq 'dnf-command(config-manager)' jq
+
 MAX_NUMBER_OF_RELEASES=32
+LAST_VESPA_7_RELEASE="7.594.36"
 
 # Cloudsmith repo
 rpm --import 'https://dl.cloudsmith.io/public/vespa/open-source-rpms/gpg.0F3DA3C70D35DA7B.key'
-curl -1sLf 'https://dl.cloudsmith.io/public/vespa/open-source-rpms/config.rpm.txt?distro=el&codename=8' > /tmp/vespa-open-source-rpms.repo
+curl  --silent --show-error --location --fail \
+  --tlsv1 --output /tmp/vespa-open-source-rpms.repo \
+  'https://dl.cloudsmith.io/public/vespa/open-source-rpms/config.rpm.txt?distro=el&codename=8'
 dnf config-manager --add-repo '/tmp/vespa-open-source-rpms.repo'
 rm -f /tmp/vespa-open-source-rpms.repo
 
-# Allow the last Vespa 7 release to remain in the repo
-VERSIONS_TO_DELETE=$(dnf list -y --quiet --showduplicates --disablerepo='*' --enablerepo=vespa-open-source-rpms vespa | awk '/[0-9].*\.[0-9].*\.[0-9].*/{print $2}' | sort -V | grep -v "7.594.36" | head -n -$MAX_NUMBER_OF_RELEASES)
+VESPA_VERSIONS=$(dnf list -y --quiet --showduplicates --disablerepo='*' --enablerepo=vespa-open-source-rpms vespa || true)
+if [[ -z "$VESPA_VERSIONS" ]]; then
+  echo "No Vespa versions found, nothing to do. Exiting."
+  exit 0
+fi
 
+# Note: Allow the last Vespa 7 release to remain in the repo!
+VERSIONS_TO_DELETE=$(echo "${VESPA_VERSIONS}" | awk '/[0-9].*\.[0-9].*\.[0-9].*/{print $2}' | sort -V | grep -v "${LAST_VESPA_7_RELEASE}" | head -n -$MAX_NUMBER_OF_RELEASES)
```

Comment:
Over-generalized regex - you probably mean `/[0-9]+\.[0-9]+\.[0-9]+/` ?

Reply:
I just copied the existing regex and indeed I like your suggestion better. 

I'll tell you what.. Given we want to handle the 8+ only, I'll give you one even better:

```suggestion
VERSIONS_TO_DELETE=$(echo "${VESPA_VERSIONS}" | awk '/[8-9]\.[0-9]+\.[0-9]+/{print $2}' | sort -V | head -n -$MAX_NUMBER_OF_RELEASES)
```

Reply:
Sweet